cmake/OpenCVCompilerOptions.cmake

```diff
@@ -377,10 +377,15 @@ if(MSVC)
 endif()
 
 if(NOT OPENCV_FP16_DISABLE)
+  if(ARM)
+    set(FP16_OPTION "-mfpu=neon-fp16")
+  elseif(X86 OR X86_64)
+    set(FP16_OPTION "-mf16c")
+  endif()
```

Comment:
I believe `try_compile` is preferable here. `try_run` doesn't work in case of cross-compilation.